_blog/2024-01-24-using-cronjobs-rust.mdx

```diff
@@ -0,0 +1,294 @@
+thumb: 
+cover: 
+date: '2024-01-23T14:30:00'
+---
+
+In this article we’re going to talk about how you can write your own cron jobs as a web service using Shuttle! 
+
+Cron jobs (or “scheduled tasks”) are useful for many things. They allow you to automatically do things like:
+
+- Automate data back-ups
+- Adding daily reminders (for example to customers who are signed up to a service you own but haven’t started using it yet)
+- Creating/writing reports
+
+If you’re interested in the final code and want to deploy quickly, you can find the repo [here.](https://github.com/joshua-mo-143/shuttle-cron-template) You can deploy it with two simple steps:
+
+1) Run `cargo shuttle init --from joshua-mo-143/shuttle-cron-template` and follow the prompt (requires `cargo-shuttle` installed)
+
+2) Run `cargo shuttle deploy`. That’s it!
+
+## Getting Started
+
+If you don’t have `cargo-shuttle` installed, you can install it by running the command below (requires Rust + Cargo installed):
+
+```bash
+cargo install cargo-shuttle
+```
+
+To get started, you’ll want to initialise a project with `cargo-shuttle`:
+
+```bash
+cargo shuttle init 
+```
+
+You’ll want to make sure to follow the prompt and choose `None` when the framework options come up. This will spawn a custom service that you can use with Shuttle. For this article, we will be using the project name `shuttle-example-cron`. 
+
+Once done, we’ll want to install the following dependencies:
+
+- `serde` - Allows us to de/serialize tasks
+- `chrono` - Allows us to use timestamps
+- `apalis` - The task queue framework we’ll be using
+- `sqlx` - Allows you to interact with your provisioned database
+- `shuttle-shared-db` - Provisions database for you from Shuttle (locally via Docker, via the runtime in deployment)
+- `tower` - Used with apalis so that the cron job can be hosted.
+
+We can install all of these with the following shell snippet:
+
+```bash
+cargo add apalis -F cron,postgres,extensions,retry
+cargo add chrono -F serde,clock
+cargo add serde -F derive
+cargo add shuttle-shared-db -F postgres
+cargo add sqlx -F runtime-tokio-native-tls,postgres
+cargo add tower 
+```
+
+Now that everything is installed, it’s time to get coding!
+
+## Writing our first cron job
+
+### Adding a database
+
+Before we do anything else, let’s provision a database using the Shuttle runtime. You’ll want to add the `shuttle-shared-db` annotation to your main function like so:
+
+```rust
+use sqlx::PgPool;
+
+pub struct MyService {
+    db: PgPool,
+}
+
+#[shuttle_runtime::main]
+async fn shuttle_main(
+    #[shuttle_shared_db::Postgres] db: PgPool,
+    ) -> Result<MyService, shuttle_runtime::Error> {
+    Ok(MyService { db })
+}
+```
+
+As you can see, pretty simple! You would otherwise need to run a Docker command to spin this up and in deployment you would also need to manually create your Postgres instance or use Terraform (or some other kind of platform that utilises infrastructure files).
```

Comment:
WDYM by "and in deployment" ? If you're talking about production, I would suggest something like "and for production you would need to instantiate and manage your Postgres instance or rely on an IaC (Infrastructure-as-Code) tool like Terraform to do it".